lib/open_mic.rb

```diff
@@ -0,0 +1,39 @@
+class OpenMic
+  attr_reader :attr, :location, :date, :performers, :comp
+
+  def initialize(attr = {})
+    @attr = attr
+    @location = attr[:location]
+    @date = attr[:date]
+    @performers = []
+    @comp = {}
+    # require "pry"; binding.pry
+  end
+
+  def welcome(user)
+    @performers << user
+    @comp["#{user.name} jokes"] = [] if comp[user.name].nil?
+    @comp["#{user.name} jokes"] << user.jokes
+  end
+
+  def duplicate_count(array) # researched online and found this. Tinkered with it
+    #to work with my code but it is not mine.
+    array.each_with_object(Hash.new(0)) do |value, hash|
+      hash[value] += 1
+    end.count do |(value,count)|
+      count > 1
+    end
+  end
+
+  def all_jokes
+    all_jokes = comp.values
+    all_jokes.flatten!
+    all_jokes
+  end
+
+  def repeated_jokes?
```

Comment:
I really like that you created a helper method to get all jokes - that is a great separation of responsibilities.  

Once you have a list of all jokes, you can simplify the duplication check by using the `.length`.  If the length of `all_jokes` is the same as the length of `all_jokes.uniq`, then we know there are no duplicates.